skate-plugin/project-gen/src/jvmMain/kotlin/slack/tooling/aibot/ChatPanel.kt

```diff
@@ -15,16 +15,31 @@
  */
 package slack.tooling.aibot
 
+import androidx.compose.runtime.Composable
 import androidx.compose.ui.awt.ComposePanel
+import com.slack.circuit.foundation.Circuit
+import com.slack.circuit.foundation.CircuitContent
 import java.awt.Dimension
 import javax.swing.JComponent
+import slack.tooling.aibot.ChatWindowUi.ChatWindow
 import slack.tooling.projectgen.SlackDesktopTheme
 
 object ChatPanel {
   fun createPanel(): JComponent {
     return ComposePanel().apply {
       preferredSize = Dimension(400, 600)
-      setContent { SlackDesktopTheme { ChatWindow() } }
+      setContent { SlackDesktopTheme { ChatApp() } }
     }
   }
+
+  @Composable
+  private fun ChatApp() {
+    val circuit =
+      Circuit.Builder()
+        .addPresenter<ChatScreen, ChatScreen.State>(ChatPresenter())
+        .addUi<ChatScreen, ChatScreen.State> { state, modifier -> ChatWindow(state, modifier) }
+        .build()
```

Comment:
you should wrap this in a `remember { }` block, otherwise it'll get recreated every time

Reply:
2d42f95f8a9a4c4320581950aabb065bc1516b8d in ChatPanel.kt

Reply:
f27ddc1af95d1c0b90e33622407588cd464def09 under `ChatPanel.kt`